### PR TITLE
M3 migration with custom theme

### PR DESCRIPTION
**Pull request by OwenScott-GDS**

### CODE REVIEW

File: app/src/main/java/com/example/reply/data/local/LocalEmailsDataProvider.kt [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        ),
+        Email(
+            3L,
+            LocalAccountsDataProvider.getContactAccountById(8L),
+            listOf(LocalAccountsDataProvider.defaultAccount),
+            "High school reunion?",
+            """
+                Hi friends,
+                I was at the grocery store on Sunday night.. when I ran into Genie Williams! I almost didn't recognize her afer 20 years!
+                Anyway, it turns out she is on the organizing committee for the high school reunion this fall. I don't know if you were planning on going or not, but she could definitely use our help in trying to track down lots of missing alums. If you can make it, we're doing a little phone-tree party at her place next Saturday, hoping that if we can find one person, thee more will...
```
Comment (yenerm) [inline app/src/main/java/com/example/reply/data/local/LocalEmailsDataProvider.kt:54]: ditto

File: app/src/main/java/com/example/reply/data/local/LocalEmailsDataProvider.kt [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+            createdAt = "2 hours ago",
+            mailbox = MailboxType.Sent,
+        ),
+        Email(
+            id = 4L,
+            sender = LocalAccountsDataProvider.getContactAccountById(11L),
+            subject = "Brazil trip",
+            body = """
+                Thought we might be able to go over some details about our upcoming vacation.
+                I've been doing a bit of research and have come across a few paces in Northern Brazil that I think we should check out. One, the north has some of the most predictable wind on the planet. I'd love to get out on the ocean and kitesurf for a couple of days if we're going to be anywhere near or around Taiba. I hear it's beautiful there and if you're up for it, I'd love to go. Other than that, I haven't spent too much time looking into places along our road trip route. I'm assuming we can find places to stay and things to do as we drive and find places we think look interesting. But... I know you're more of a planner, so if you have ideas or places in mind, lets jot some ideas down!
```
Comment (yenerm) [inline app/src/main/java/com/example/reply/data/local/LocalEmailsDataProvider.kt:65]: same, check other instances as well.

File: app/src/main/java/com/example/reply/ui/ReplyUiState.kt [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import com.example.reply.data.MailboxType
+import com.example.reply.data.local.LocalEmailsDataProvider
+
+data class ReplyUiState(
+    val mailboxes: Map<MailboxType, List<Email>> = emptyMap(),
+    val currentMailbox: MailboxType = MailboxType.Inbox,
+    val currentSelectedEmail: Email = LocalEmailsDataProvider.defaultEmail,
+    val isShowingHomepage: Boolean = true
+) {
+    val currentMailboxEmails: List<Email> by lazy { mailboxes[currentMailbox]!! }
```
Comment (yenerm) [inline app/src/main/java/com/example/reply/ui/ReplyUiState.kt:13]: just checking this line is no new and previously SME approved?

**Review by yenerm — CHANGES_REQUESTED** [review]
Meghan to review M3 changes in more detail

File: app/src/main/java/com/example/reply/ui/ReplyDetailsScreen.kt [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+            verticalArrangement = Arrangement.Center
+        ) {
+            Text(
+                text = email.sender.firstName,
+                style = MaterialTheme.typography.labelMedium
+            )
+            Text(
+                text = email.createdAt,
+                style = MaterialTheme.typography.labelMedium,
+                color = MaterialTheme.colorScheme.outline
```
Comment (MagicalMeghan) [inline app/src/main/java/com/example/reply/ui/ReplyDetailsScreen.kt:243]: i see you are setting a lot of colors in this app, curious why you aren't using the default colors for components

File: app/src/main/java/com/example/reply/ui/ReplyHomeContent.kt [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun ReplyEmailListItem(
+    email: Email,
+    selected: Boolean,
+    onCardClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Card(
+        modifier = modifier.padding(
```
Comment (MagicalMeghan) [inline app/src/main/java/com/example/reply/ui/ReplyHomeContent.kt:105]: pass in from parent

File: app/src/main/java/com/example/reply/ui/ReplyHomeContent.kt [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+                color = MaterialTheme.colorScheme.onSurfaceVariant,
+                overflow = TextOverflow.Ellipsis
+            )
+        }
+    }
+}
+
+@Composable
+private fun ReplyEmailItemHeader(email: Email, modifier: Modifier = Modifier) {
+    Row(modifier = modifier.fillMaxWidth()) {
```
Comment (MagicalMeghan) [inline app/src/main/java/com/example/reply/ui/ReplyHomeContent.kt:141]: pass in from parent

File: app/src/main/java/com/example/reply/ui/ReplyHomeContent.kt [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+}
+
+@Composable
+fun ReplyProfileImage(
+    @DrawableRes drawableResource: Int,
+    description: String,
+    modifier: Modifier = Modifier,
+) {
+    Image(
+        modifier = modifier.clip(CircleShape),
```
Comment (MagicalMeghan) [inline app/src/main/java/com/example/reply/ui/ReplyHomeContent.kt:176]: pass in from parent

File: app/src/main/java/com/example/reply/ui/ReplyHomeContent.kt [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+        painter = painterResource(drawableResource),
+        contentDescription = description,
+    )
+}
+
+@Composable
+fun ReplyLogo(
+    modifier: Modifier = Modifier,
+    color: Color = MaterialTheme.colorScheme.primary
```
Comment (MagicalMeghan) [inline app/src/main/java/com/example/reply/ui/ReplyHomeContent.kt:185]: is color optional? if not it should be before modifier

File: app/src/main/java/com/example/reply/ui/ReplyHomeContent.kt [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+    )
+}
+
+@Composable
+private fun ReplyHomeTopBar(modifier: Modifier = Modifier) {
+    Row(
+        horizontalArrangement = Arrangement.SpaceBetween,
+        verticalAlignment = Alignment.CenterVertically,
+        modifier = modifier
+            .fillMaxWidth()
```
Comment (MagicalMeghan) [inline app/src/main/java/com/example/reply/ui/ReplyHomeContent.kt:201]: pass in from parent

File: app/src/main/java/com/example/reply/ui/ReplyHomeScreen.kt [hunk 10]
```diff
@@ -0,0 +1,286 @@
...
+                onClick = { onTabPressed(navItem.mailboxType) }
+            )
+        }
+    }
+}
+
+@Composable
+private fun NavigationDrawerHeader(modifier: Modifier) {
+    Row(
+        modifier = modifier
```
Comment (MagicalMeghan) [inline app/src/main/java/com/example/reply/ui/ReplyHomeScreen.kt:266]: pass in from parent

File: app/src/main/java/com/example/reply/ui/ReplyHomeScreen.kt [hunk 11]
```diff
@@ -0,0 +1,286 @@
...
+@Composable
+private fun NavigationDrawerContent(
+    selectedDestination: MailboxType,
+    onTabPressed: ((MailboxType) -> Unit),
+    navigationItemContentList: List<NavigationItemContent>,
+    modifier: Modifier = Modifier
+) {
+    Column(
+        modifier
+            .wrapContentWidth()
```
Comment (MagicalMeghan) [inline app/src/main/java/com/example/reply/ui/ReplyHomeScreen.kt:233]: pass in from parent

File: app/src/main/java/com/example/reply/ui/ReplyHomeScreen.kt [hunk 12]
```diff
@@ -0,0 +1,286 @@
...
+private fun ReplyAppContent(
+    navigationType: ReplyNavigationType,
+    contentType: ReplyContentType,
+    replyUiState: ReplyUiState,
+    onTabPressed: ((MailboxType) -> Unit),
+    onEmailCardPressed: (Email) -> Unit,
+    navigationItemContentList: List<NavigationItemContent>,
+    modifier: Modifier = Modifier,
+) {
+    Row(modifier = modifier.fillMaxSize()) {
```
Comment (MagicalMeghan) [inline app/src/main/java/com/example/reply/ui/ReplyHomeScreen.kt:136]: pass in from parent

File: app/src/main/java/com/example/reply/ui/ReplyHomeScreen.kt [hunk 13]
```diff
@@ -0,0 +1,286 @@
...
+            drawerContent = {
+                PermanentDrawerSheet(Modifier.width(dimensionResource(R.dimen.drawer_width))) {
+                    NavigationDrawerContent(
+                        selectedDestination = replyUiState.currentMailbox,
+                        onTabPressed = onTabPressed,
+                        navigationItemContentList = navigationItemContentList
+                    )
+                }
+            },
+            modifier = Modifier.testTag(navigationDrawerContentDescription)
```
Comment (MagicalMeghan) [inline app/src/main/java/com/example/reply/ui/ReplyHomeScreen.kt:92]: Code is a little hard to follow, but shouldnt the composable modifier be passed in here?

File: app/src/main/java/com/example/reply/data/Account.kt [hunk 14]
```diff
@@ -0,0 +1,22 @@
+package com.example.reply.data
```
Comment (yenerm) [inline app/src/main/java/com/example/reply/data/Account.kt:1]: No license?!

File: app/src/main/java/com/example/reply/data/Email.kt [hunk 15]
```diff
@@ -0,0 +1,25 @@
+package com.example.reply.data
```
Comment (yenerm) [inline app/src/main/java/com/example/reply/data/Email.kt:1]: missing license, applies to other files as well

**Review by yenerm — CHANGES_REQUESTED** [review]
missing licenses

File: app/src/main/AndroidManifest.xml [hunk 16]
```diff
@@ -1,24 +1,11 @@
 <?xml version="1.0" encoding="utf-8"?>
-<!--
```
Comment (yenerm) [inline app/src/main/AndroidManifest.xml:2]: can you double check licenses again?

**Review by yenerm — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/reply/ui/ReplyDetailsScreen.kt [hunk 17]
```diff
@@ -0,0 +1,306 @@
...
+            Text(
+                text = stringResource(replyUiState.currentSelectedEmail.subject),
+                style = MaterialTheme.typography.titleMedium,
+                color = MaterialTheme.colorScheme.onSurfaceVariant
+            )
+        }
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
```
Comment (MagicalMeghan) [inline app/src/main/java/com/example/reply/ui/ReplyDetailsScreen.kt:133]: add experimentalmaterial3 in the build.gradle file

File: app/src/main/java/com/example/reply/ui/ReplyHomeContent.kt [hunk 18]
```diff
@@ -0,0 +1,247 @@
...
+fun ReplyEmailListItem(
+    email: Email,
+    selected: Boolean,
+    onCardClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Card(
+        modifier = modifier,
+        colors = CardDefaults.cardColors(
+            containerColor = if (selected)
```
Comment (MagicalMeghan) [inline app/src/main/java/com/example/reply/ui/ReplyHomeContent.kt:133]: if/else looks strange here

File: app/src/main/java/com/example/reply/ui/ReplyHomeScreen.kt [hunk 19]
```diff
@@ -0,0 +1,316 @@
...
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import com.example.reply.R
+import com.example.reply.data.Email
+import com.example.reply.data.MailboxType
+import com.example.reply.data.local.LocalAccountsDataProvider
+import com.example.reply.ui.utils.ReplyContentType
+import com.example.reply.ui.utils.ReplyNavigationType
+
+@OptIn(ExperimentalMaterial3Api::class)
```
Comment (MagicalMeghan) [inline app/src/main/java/com/example/reply/ui/ReplyHomeScreen.kt:63]: add experimentalmaterial3api into the gradle

File: app/src/main/java/com/example/reply/ui/ReplyDetailsScreen.kt [hunk 20]
```diff
@@ -0,0 +1,306 @@
...
+                        Modifier
+                            .fillMaxWidth()
+                            .padding(bottom = dimensionResource(R.dimen.detail_topbar_padding_bottom))
+                    )
+                }
+                ReplyEmailDetailsCard(
+                    email = replyUiState.currentSelectedEmail,
+                    mailboxType = replyUiState.currentMailbox,
+                    isFullScreen = isFullScreen,
+                    modifier = if (isFullScreen)
```
[the remark this answers is not in the document]
Comment (MagicalMeghan) [reply app/src/main/java/com/example/reply/ui/ReplyDetailsScreen.kt:86]: [resolved over chat]

**Review by MagicalMeghan — APPROVED** [review]
